PWGCF/Flow/Tasks/flowAnalysisGF.cxx

```diff
@@ -0,0 +1,686 @@
+        // reject events affected by the ITS ROF border
+        // https://its.cern.ch/jira/browse/O2-4309
+        return 0;
+      }
+      registry.fill(HIST("hEventCount"), 4.5);
+
+      if (!collision.selection_bit(o2::aod::evsel::kNoSameBunchPileup)) {
+        // rejects collisions which are associated with the same "found-by-T0" bunch crossing
+        // https://indico.cern.ch/event/1396220/#1-event-selection-with-its-rof
+        return 0;
+      }
+      registry.fill(HIST("hEventCount"), 5.5);
+
+      if (!collision.selection_bit(o2::aod::evsel::kIsGoodZvtxFT0vsPV)) {
+        // removes collisions with large differences between z of PV by tracks and z of PV from FT0 A-C time difference
+        // use this cut at low multiplicities with caution
+        return 0;
+      }
+      registry.fill(HIST("hEventCount"), 6.5);
+
+      if (!collision.selection_bit(o2::aod::evsel::kIsVertexITSTPC)) {
+        // selects collisions with at least one ITS-TPC track, and thus rejects vertices built from ITS-only tracks
+        return 0;
+      }
+      registry.fill(HIST("hEventCount"), 7.5);
+    }
+
+    float vtxz = -999;
+    if (collision.numContrib() > 1) {
+      vtxz = collision.posZ();
+      float zRes = TMath::Sqrt(collision.covZZ());
+      if (zRes > 0.25 && collision.numContrib() < 20)
+        vtxz = -999;
+    }
+    // auto multV0A = collision.multFV0A();
+    // auto multT0A = collision.multFT0A();
+    // auto multT0C = collision.multFT0C();
+    auto multNTracksPV = collision.multNTracksPV();
+
+    if (vtxz > vtxZup || vtxz < vtxZlow)
+      return 0;
+    if (multNTracksPV < fMultPVCutLow->Eval(centrality))
+      return 0;
+    if (multNTracksPV > fMultPVCutHigh->Eval(centrality))
+      return 0;
+    if (multTrk < fMultCutLow->Eval(centrality))
+      return 0;
+    if (multTrk > fMultCutHigh->Eval(centrality))
+      return 0;
+    registry.fill(HIST("hEventCount"), 8.5);
+
+    return 1;
+  }
+
+  template <typename TTrack>
+  bool trackSelected(TTrack track, const int& field)
+  {
+    double phimodn = track.phi();
+    if (field < 0) // for negative polarity field
+      phimodn = TMath::TwoPi() - phimodn;
+    if (track.sign() < 0) // for negative charge
+      phimodn = TMath::TwoPi() - phimodn;
+    if (phimodn < 0)
+      LOGF(warning, "phi < 0: %g", phimodn);
+
+    phimodn += TMath::Pi() / 18.0; // to center gap in the middle
+    phimodn = fmod(phimodn, TMath::Pi() / 9.0);
+    registry.fill(HIST("pt_phi_bef"), track.pt(), phimodn);
+    if (phimodn < fPhiCutHigh->Eval(track.pt()) && phimodn > fPhiCutLow->Eval(track.pt()))
+      return false; // reject track
+    if (cfgDoubleTrackFunction) {
+      if (phimodn < fPhiCutHigh2->Eval(track.pt()) && phimodn > fPhiCutLow2->Eval(track.pt()))
+        return false; // reject track
+    }
+    registry.fill(HIST("pt_phi_aft"), track.pt(), phimodn);
+    return true;
+  }
+
+  enum datatype {
+    kReco,
+    kGen
+  };
+
+  template <datatype dt>
```

Comment:
Any reason for `dt` being a template parameter and not just a function parameter?

Reply:
I copied it from the flowGenericFramework task, where a datatype is used. I think it is because dt is either kGen or kReco and when using it in this way it is always clear which one is used in the different process functions.